packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -435,7 +435,11 @@ export class CdkToolkit {
   /**
    * Validate the stacks for errors and warnings according to the CLI's current settings
    */
-  private async validateStacks(stacks: StackCollection) {
+  private async validateStacks(stacks: StackCollection, stackNames: string[]) {
+    if (stacks.stackCount == 0) {
```

Comment:
I agree with this check, but I think it should not be part of this function. Validating that there are stacks is not what this function is doing (it's checking stack metadata).

How about adding a new function?

```ts
function validateStacksSelected(stacks: StackCollection, stackNames: string[]);
```

Or something?